### DISCUSSION TITLE
How do I push commits to a local repository with an SSH key?

### DISCUSSION THREAD

**Discussion by jmkd3v** [gitpython-developers/GitPython#1203] [Q&A] [answered]
I have a local repository that I've already cloned and I want to push changes to remote (GitHub) from a different account.  
I've tried something like this:
```py
id_file = os.path.expanduser('~/.ssh/id_rsa_api')
id_cmd = "ssh -i " + id_file
repo = Repo(path)
with repo.git.custom_environment(GIT_SSH_COMMAND=id_cmd):
    # rest of code here
```
but it's had no effect - the commits always are pushed under my main GiitHub account, no matter what SSH key I use.

**Comment by Byron** [comment]
Did the GIT_SSH_COMMAND environment variable have an effect when running `git` on the command-line directly?

From the code I see here I would expect this to be picked up when interacting with remotes.

**Reply by jmkd3v** [reply]
Odd - it does not apply in the command line either. Maybe I should remove my main account from Git and try again?

**Reply by jmkd3v** [reply]
Update: I was able to solve this problem by deleting my GitHub credentials and then changing my email and username on the local level.

**Comment by Byron** [comment, marked as answer]
@jmkd3v wrote
> Update: I was able to solve this problem by deleting my GitHub credentials and then changing my email and username on the local level.
